### DISCUSSION TITLE
Disable for specific filetypes (Such as vim.fugitive)

### DISCUSSION THREAD

**Discussion by pixeljae** [shortcuts/no-neck-pain.nvim#228] [Q&A]
Noob question, I have the option `enableOnVimEnter` set to true since the plugin is great for pretty much everything.  However, I would like to be able to disable it when using the vertical split option on certain filetypes (Such as vim.fugitive).

How can I accomplish this? Any help is appreciated. Thanks for making this plugin, it really lives up to its name!

**Comment by shortcuts** [comment]
Hey, thanks a lot for the feedback, really appreciate you like the plugin :)

I can't think of a way to make it work natively **only for certain filetypes**, however I could implement an option to be able to close padding windows when entering a certain filetype (TBC if it make sense)

Are the following steps a correct use case?
1. use vim with NNP enabled
2. call something that opens a new window in vsplit (e.g. fugitive)
3. padding windows are resized, but you'd like them to be closed
4. reopens when closing this specific vsplit (e.g. fugitive)

----

an easy workaround in the meantime would be to implement an [`autocmd`](https://neovim.io/doc/user/autocmd.html) on your side for the fugitive filetype, which triggers on `WinEnter, WinClosed` and calls `:NoNeckPain`

**Reply by pixeljae** [reply]
Thanks for the quick reply.  That's actually the exact usecase.  I tried setting something like `vim.cmd([[autocmd FileType fugitive NoNeckPain]])`, but it seems to break the window handling by getting rid of left-side padding. even on the main editor window.  Attached screens for reference

How it looks with NNP enabled on all buffers
<img width="2500" alt="SCR-20230508-brgo" src="https://user-images.githubusercontent.com/6393886/236691236-5518e493-52c1-4543-9f24-516b7c53e6d6.png">

How it looks once the autocmd runs, breaks left side padding on all windows.
<img width="2521" alt="SCR-20230508-brjt" src="https://user-images.githubusercontent.com/6393886/236691273-e6a93f53-6d6e-47ea-b253-908e2a539d4a.png">

I couldn't figure out how to run the command on both WinEnter, WinClosed events, while still only disabling the plugin on the fugitive filetype. In any case, any help is appreciated but by no means is this a big issue since the plugins works fantastically regardless.

**Reply by h0i5** [reply]
Hey, did you manage to fix it? Im having the same problem with NvimTree :D
